### PR TITLE
WRKLDS-1431: e2e: migrate DCs to Deployments

### PR DESCRIPTION
**Pull request by ingvagabund**
Migrated:
- `[Serial] [sig-auth][Feature:OAuthServer] [RequestHeaders] [IdP] test RequestHeaders IdP [apigroup:config.openshift.io][apigroup:user.openshift.io]`: works without DC API enabled
- `[sig-builds][Feature:Builds][volumes] ...`: migrating with triggers
- `[sig-cli] oc --request-timeout works as expected`: creating a deployment instead of a DC
- `[sig-cli] oc adm build-chain [apigroup:build.openshift.io][apigroup:image.openshift.io][apigroup:project.openshift.io]`: migrating with triggers
- `[sig-cli] oc builds ...`:  migrating with triggers
- `[sig-cli] oc debug ensure debug does not depend on a container actually existing for the selected resource`: duplicating for deployments
- `[sig-cli] oc env can set environment variables`:  migrating with triggers
- `[sig-cli] oc idle [apigroup:apps.openshift.io][apigroup:route.openshift.io][apigroup:project.openshift.io][apigroup:image.openshift.io] by ...`: duplicated and changed for deployments
- `[sig-cli] oc run can use --image flag correctly`: creating a deployment instead of a DC
- `[sig-cli] oc set image can set images for pods and deployments`: migrating with triggers
- `[sig-cli] oc status returns expected help messages`: works without DC as long as --suggest arg is provided
- `[sig-cli] templates process`: migrating with triggers
- `[sig-devex][Feature:Templates] templateinstance readiness test `: migrating with triggers
- `[sig-devex][Feature:ImageEcosystem][perl][Slow] hot deploy for openshift perl image hot deploy test should work [apigroup:image.openshift.io][apigroup:operator.openshift.io][apigroup:config.openshift.io][apigroup:build.openshift.io][apigroup:apps.openshift.io]`: migrating with triggers + replacing all the waiting code and Deployment -> RS -> pod retrival code.
- `[sig-imageregistry][Feature:ImageLookup] Image policy should update OpenShift object image fields when local names are on`:
- `[sig-network-edge][Feature:Idling] Unidling ...`: simple DC replacement (no triggers)
- `[sig-scheduling][Early] The openshift-apiserver pods [apigroup:apps.openshift.io][apigroup:authorization.openshift.io][apigroup:build.openshift.io][apigroup:image.openshift.io][apigroup:project.openshift.io][apigroup:quota.openshift.io][apigroup:route.openshift.io][apigroup:security.openshift.io][apigroup:template.openshift.io] should be scheduled on different nodes`: already without DC dependency, just removing the apigroup.

Ignoring:
- `[sig-imageregistry][Feature:ImageLookup] Image policy should update OpenShift object image fields when local names are on [apigroup:image.openshift.io][apigroup:apps.openshift.io]`: testing DCs
- `[sig-imageregistry][Feature:ImagePrune][Serial][Suite:openshift/registry/serial][Local] Image hard prune [apigroup:apps.openshift.io][apigroup:user.openshift.io] should delete orphaned blobs [apigroup:image.openshift.io]`: not running at all, fails on missing dc/docker-registry.
- `[sig-imageregistry][Feature:ImagePrune][Serial][Suite:openshift/registry/serial][Local] Image hard prune [apigroup:apps.openshift.io][apigroup:user.openshift.io] should show orphaned blob deletions in dry-run mode [apigroup:image.openshift.io]`: not running at all, fails on missing dc/docker-registry.
- `[sig-devex][Feature:ImageEcosystem][mariadb][Slow] openshift mariadb image Creating from a template should instantiate the template [apigroup:image.openshift.io][apigroup:operator.openshift.io][apigroup:config.openshift.io][apigroup:apps.openshift.io]`: broken
- `[sig-devex][Feature:ImageEcosystem][mysql][Slow] openshift mysql image Creating from a template should instantiate the template [apigroup:apps.openshift.io]`: broken
- `[sig-cli] oc probe can ensure the probe command is functioning as expected on deploymentconfigs [apigroup:apps.openshift.io]`: testing DCs
- `oc explain should contain spec+status for apps.openshift.io [apigroup:apps.openshift.io]`: testing DCs
- `oc explain should contain proper fields description for apps.openshift.io [apigroup:apps.openshift.io]`: testing DCs
- `[sig-cli] oc basics can create deploymentconfig and clusterquota [apigroup:apps.openshift.io]`: testing DCs
- `[sig-scalability][Feature:Performance] Load cluster should populate the cluster [Slow][Serial]`: broken by https://github.com/openshift/origin/pull/26072 as the e2e is expected to be [run separately](https://github.com/openshift/origin/tree/master/test/extended/cluster#running-cluster-mirror): The following error is produced:
  ```
   [sig-scalability][Feature:Performance] Load cluster [It] should populate the cluster [Slow][Serial][apigroup:template.openshift.io][apigroup:build.openshift.io]
  github.com/openshift/origin/test/extended/cluster/cl.go:63

    [FAILED] Unable to process template file. Error: Error running /usr/bin/oc process -f /home/jchaloup/Projects/src/github.com/openshift/origin/test/extended/testdata/cluster/quickstarts/cakephp-mysql.json -p IDENTIFIER=0:
    StdOut>

    StdErr>
    error: unknown parameter name "IDENTIFIER"
    exit status 1
    In [It] at: github.com/openshift/origin/test/extended/cluster/utils.go:529 @ 07/30/24 13:07:16.067
  ------------------------------
  ```
  IDENTIFIER was removed by https://github.com/openshift/origin/pull/26072.
- `[sig-scalability]`: can't find any run in https://search.dptools.openshift.org/?search=%5C%5Bsig-scalability%5C%5D&maxAge=48h&context=1&type=junit&name=&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job

### CODE REVIEW

**Comment by ingvagabund** [comment]
> ci/prow/e2e-gcp-ovn-builds

Based on https://prow.ci.openshift.org/job-history/gs/test-platform-results/pr-logs/directory/pull-ci-openshift-origin-master-e2e-gcp-ovn-builds the job has been failing for other PRs as well.

**Comment by JoelSpeed** [comment]
If we are no longer testing DCs, how do we ensure that the customers who are relying on DCs do not get broken as we move towards future releases?

**Comment by dgoodwin** [comment]
For the tests renamed to remove the apps.openshift.io apigroup, these need rename records in the mapping repo so that component readiness can continue to compare their results with past releases. https://github.com/openshift-eng/ci-test-mapping should have instructions.

We do need to understand where and how DCs continue to be tested per Joel's comment. You mention there are tests that require apps.openshift.io installed, what test(s) is that and how would they NOT fail on default install? TRT couldn't quite piece together how this will continue testing.

Also would recommend you run the 4.17 nightly blocking with /payload command on the other PR for this to actually disable, once this merges.

**Comment by ingvagabund** [comment]
Lemme get back to you both once I have an answer for how DCs will continue to be tested

**Comment by ingvagabund** [comment]
https://github.com/openshift/origin/tree/master/test/extended/controller_manager is a home for [sig-apps][Feature:DeploymentConfig]. Also, https://github.com/openshift/origin/blob/master/test/extended/testdata/cmd/test/cmd/deployments.sh is going through various cases for deployments as well. The DC coverage is still present.

**Comment by ingvagabund** [comment]
@JoelSpeed's response from https://github.com/openshift/api/pull/1954:

> Having an all capabilities test makes sense to me, but we need to make sure that something in origin is still exercising the DCs, else that will be pointless. Not sure how to make those tests only run on those particular clusters but I'm sure there's probably precedent for that.

**Comment by ingvagabund** [comment]
> We do need to understand where and how DCs continue to be tested per Joel's comment. You mention there are tests that require apps.openshift.io installed, what test(s) is that and how would they NOT fail on default install? TRT couldn't quite piece together how this will continue testing.

There's an ongoing discussion in https://redhat-internal.slack.com/archives/C032HSVS71T/p1722529302986619. The current conclusion is workloads QE will create a periodic job with all capabilities enabled. That will cover the case where we will still keep exercising DCs.

> You mention there are tests that require apps.openshift.io installed, what test(s) is that and how would they NOT fail on default install?

These are all the tests that have [apigroup:apps..openshift.io] in their name. E.g.:
- https://github.com/openshift/origin/tree/master/test/extended/controller_manager is a home for `[sig-apps][Feature:DeploymentConfig]`.
- https://github.com/openshift/origin/blob/master/test/extended/testdata/cmd/test/cmd/deployments.sh is going through various cases for deployments as well.

The still existing tests cover the following cases:
- defaults when creating a DC
- scaling DCs (using the scale subresource)
- triggers (manual, image change, config change)
- creation of deployer pull secrets
- usual CRUD operators
- setting/getting env
- managing RCs
- template processing
- rollouts

**Comment by kasturinarra** [comment]
I spoke to jan this evening and i understand that workload qe team will setup new periodic jobs and maintain these from 4.17 to test this particular epic. Setting up these jobs will be a collaborative task between openshift qe and workloads qe and this has nothing to with the jobs and profiles that dev has or they look into. Also workloads team will continue to test deploymentconfigs and make sure no tests with deploymentconfigs fail after this feature is merged and continue to have dc tests in our test suite.

**Comment by JoelSpeed** [comment]
Will it be possible to add any testing as payload blocking? Would be good to make sure we don't accept payloads where we are regressing on features, keep me honest @dgoodwin, would you also expect something blocking to make sure we don't regress?

**Comment by ingvagabund** [comment]
Can a periodic job be considered blocking? Maybe, we need to add a new job that gets run every time a new release is created?

**Comment by JoelSpeed** [comment]
> Can a periodic job be considered blocking?

Yes, there's some config in the release repo that you can use to configure a job as informing, once you can show that it passes reliably (I think 20 consecutive builds), then you can promote to blocking, or at least, that's the process for new features, this may be a slightly different requirement

**Comment by ingvagabund** [comment]
/hold
This got moved to 4.18. I will break the PR into smaller chunks so it's easier to review/follow the changes.

**Comment by ingvagabund** [comment]
https://github.com/openshift-eng/ci-test-mapping/pull/148 updates the test mapping.

**Comment by ingvagabund** [comment]
To summarize:
- as per https://github.com/openshift/origin/pull/28957#issuecomment-2270853924 the DC and their features/functionalities are still tested and exercised in almost the same scope. What's reduced are cases where a DC is used as a mean to deploy a workload. In such cases a DC manifest is replaced with a corresponding Deployment.
- the changed test name mapping is updated in https://github.com/openshift-eng/ci-test-mapping/pull/148
- QE will own a new periodically running job that will test DC functionality even after the DC capability is disabled. The new job will be created before the DC capability gets disabled by default to guarantee the testing continues without any interruption.

**Comment by ingvagabund** [comment]
@JoelSpeed @dgoodwin the PR is ready for review once you have some time. Thx.

File: test/extended/builds/volumes.go [hunk 1]
```diff
@@ -51,7 +49,7 @@ var _ = g.Describe("[sig-builds][Feature:Builds][volumes] build volumes", func()
 			}
 		})
 
-		g.It("should mount given secrets and configmaps into the build pod for source strategy builds [apigroup:image.openshift.io][apigroup:build.openshift.io][apigroup:apps.openshift.io]", func() {
+		g.It("should mount given secrets and configmaps into the build pod for source strategy builds [apigroup:image.openshift.io][apigroup:build.openshift.io]", func() {
```
Comment (dgoodwin) [inline test/extended/builds/volumes.go:52]: Can't remember if we discussed this before but you may want to rename these tests in the test mapping repo so history carries over and component readiness can compare to past releases. Unless, you feel this is now a new test that should not be compared anymore?

"New" tests may soon be required to have a very high pass rate 99% by component readiness.

**Review by dgoodwin — COMMENTED** [review]
/approve

File: test/extended/cli/status.go [hunk 2]
```diff
@@ -19,12 +19,12 @@ var _ = g.Describe("[sig-cli] oc status", func() {
...
 	var oc = exutil.NewCLI("oc-status")
 
-	g.It("returns expected help messages [apigroup:project.openshift.io][apigroup:build.openshift.io][apigroup:image.openshift.io][apigroup:apps.openshift.io][apigroup:route.openshift.io]", func() {
+	g.It("returns expected help messages [apigroup:project.openshift.io][apigroup:build.openshift.io][apigroup:image.openshift.io][apigroup:route.openshift.io]", func() {
 		out, err := oc.Run("status").Args("-h").Output()
 		o.Expect(err).NotTo(o.HaveOccurred())
 		o.Expect(out).To(o.ContainSubstring("oc describe buildconfig"))
 
-		out, err = oc.Run("status").Output()
+		out, err = oc.Run("status").Args("--suggest").Output()
```
Comment (ardaguclu) [inline test/extended/cli/status.go:27]: What is the reason of this change?

File: test/extended/testdata/cmd/test/cmd/deployments.sh [hunk 3]
```diff
@@ -81,7 +81,7 @@ os::cmd::expect_success 'oc delete all --all'
 sleep 1
 os::cmd::expect_success 'oc delete all --all'
 
-os::cmd::expect_success 'oc process -f ${TEST_DATA}/application-template-dockerbuild.json -l app=dockerbuild | oc create -f -'
+os::cmd::expect_success 'oc process -f ${TEST_DATA}/application-template-dockerbuild-dc.json -l app=dockerbuild | oc create -f -'
```
Comment (ardaguclu) [inline test/extended/testdata/cmd/test/cmd/deployments.sh:84]: Is this test the only existence reason of application-template-dockerbuild-dc.json?. If yes, why don't we remove this test as well as this json file?

Comment (ardaguclu) [reply test/extended/testdata/cmd/test/cmd/deployments.sh:84]: As far as I understand, this is the only piece in origin referencing deploymentconfig and it is better to remove it.

**Comment by ardaguclu** [comment]
ipsec-serial is already broken and the other failures look unrelated. Dropped a few comments, overall looks good to me.

File: test/extended/testdata/cmd/test/cmd/deployments.sh (hunk 3 above)
Comment (ingvagabund) [reply test/extended/testdata/cmd/test/cmd/deployments.sh:84]: deployment.sh is still testing some of the DC functionality. I'd prefer to keep it and remove it once deployment.sh gets migrated to a go based test.

File: test/extended/cli/status.go (hunk 2 above)
Comment (ingvagabund) [reply test/extended/cli/status.go:27]: `oc status` prints a different output that does not contain "oc new-app" text when DC API is disabled. `--suggest` puts the text back.
